types/svg-parser/index.d.ts

```diff
@@ -1,14 +1,24 @@
-// Type definitions for svg-parser 1.0
+// Type definitions for svg-parser 2.0
 // Project: https://gitlab.com/Rich-Harris/svg-parser#README
 // Definitions by: mrmlnc <https://github.com/mrmlnc>
+//                 Joel Shinness <https://github.com/JoelCodes>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.1
 
-export interface Node {
-    name: string;
-    attributes: Record<string, string | number>;
-    children: Node[];
+export interface TextNode {
+    type: 'text';
+    value?: string | boolean | number;
+}
+
+export interface ElementNode {
+    type: string;
```

Comment:
As I understand from the tests, an element always has the `element` type.

https://github.com/Rich-Harris/svg-parser/blob/734f11b37397eafed09a4688a505972d9f5613b7/src/index.js#L74

Reply:
I've changed to that.  I was being a little more general, but I do like making it more specific.